packages/attachments/src/model.ts

```diff
@@ -67,6 +67,11 @@ export interface IAttachmentsModel extends IDisposable {
    */
   set(key: string, attachment: nbformat.IMimeBundle): void;
 
+  /**
+   * Remove the attachment whose name is the specified key
+   */
+  remove(key: string): void;
```

Comment:
I think this change requires a major version bump of the `attachments` package, since any alternative implementations of the interface will now be broken.

Reply:
Can that be avoided by making `remove` optional for now? I guess that would mean removing unreferenced attachments won't work until the major version bump.

Reply:
I think `remove?(key: string) : void;` seems like a good idea so we can put this in the 1.1 release.  https://stackoverflow.com/a/27523954

Reply:
I agree that making these optional, with a TODO note/issue to make the required for 2.0 would be the best middle ground. There should probably be a comment in the docstring as well then.